src/SDKs/Azure.Core/data-plane/Azure.Core/ETag.cs

```diff
@@ -13,21 +13,24 @@ namespace Azure.Core
 
         public ETag(string etag) => _ascii = Encoding.ASCII.GetBytes(etag);
 
-        public bool Equals(ETag other)
-        {
-            return this._ascii == other._ascii;
-        }
+        public bool Equals(ETag other) => _ascii.AsSpan().SequenceEqual(other._ascii);
```

Comment:
just curious: is this better performance-wise than the other implementation?

Reply:
I know there is a LINQ SequenceEqual, but it is slower (probably much slower). This method uses SIMD (vectorized) instructions to do the comparison.